src/main/java/no/stelar7/api/r4j/basic/ratelimiting/BurstRateLimiter.java

```diff
@@ -19,126 +18,162 @@ public class BurstRateLimiter extends RateLimiter
     
     private static final Logger logger = LoggerFactory.getLogger(BurstRateLimiter.class);
     
-    private ReentrantLock lock = new ReentrantLock();
-    
     public BurstRateLimiter(List<RateLimit> limits)
     {
         super(limits.toArray(new RateLimit[0]));
     }
     
     @Override
-    public void acquire()
+    public Instant acquire(String gameKey, Enum platformOrEndpoint)
     {
-        lock.lock();
+        DataCallBuilder.getLock(gameKey, platformOrEndpoint).lock();
         try
-        {
-            update();
-            long sleepTime = getDelay();
-            
-            if (sleepTime != 0)
-            {
-                Duration dur = Duration.of(sleepTime, ChronoUnit.MILLIS);
-                
-                logger.info("Ratelimited activated! Sleeping for: {}", dur);
-                logger.info("Callstack:");
-                Arrays.stream(Thread.currentThread().getStackTrace())
-                      .skip(DataCall.getCallStackSkip())
-                      .limit(DataCall.getCallStackLimit())
-                      .forEachOrdered(s -> logger.info(s.toString()));
-            }
-            
-            
-            Thread.sleep(sleepTime);
-        } catch (InterruptedException e)
-        {
-            e.printStackTrace();
+        {	
+            manageOverloadTimer(platformOrEndpoint);
+            return manageRateLimit(platformOrEndpoint);
         } finally
         {
-            lock.unlock();
+            DataCallBuilder.getLock(gameKey, platformOrEndpoint).unlock();
         }
     }
     
-    @Override
-    public void updatePermitsTakenPerX(Map<Integer, Integer> data)
-    {
-        for (Entry<Integer, Integer> key : data.entrySet())
-        {
-            for (RateLimit l : limits)
-            {
-                if (l.getTimeframeInMS() / 1000 == key.getKey())
-                {
-                    long oldVal = callCountInTime.get(l).get();
-                    long newVal = key.getValue();
-                    if (oldVal + 1 < newVal)
-                    {
-                        callCountInTime.get(l).set(newVal);
-                        logger.debug("limit {} has changed from {} to {}", key, oldVal, newVal);
-                    }
+    private void manageOverloadTimer(Enum platformOrEndpoint) {
+        long msToWait = 0;
+        
+        overloadTimerLock.lock();
+        try {
+            if(overloadTimer != 0) {
+                Instant now = Instant.now();
+                // 1s is added as security margin
+                long realtimeToWait = overloadReceivedTime.toEpochMilli() + ((overloadTimer * 1000) + 1000) - now.toEpochMilli();
```

Comment:
Since the overload timer is in seconds, we can use the plusSeconds api to update it for a bit cleaner code

```suggestion
                long realtimeToWait = overloadReceivedTime.plusSeconds(1 + overloadTimer).toEpochMilli() - now.toEpochMilli();
```